pkg/services/vmoperator/vmopmachine.go

```diff
@@ -345,7 +347,15 @@ func (v *VmopMachineService) reconcileVMOperatorVM(ctx context.Context, supervis
 		// Once the initial control plane node is ready, we can re-add the probe so
 		// that subsequent machines do not attempt to speak to a kube-apiserver
 		// that is not yet ready.
-		if infrautilv1.IsControlPlaneMachine(supervisorMachineCtx.Machine) && supervisorMachineCtx.Cluster.Status.ControlPlaneReady {
+		if v.NetworkProvider == nil {
+			return errors.Errorf(
+				"missing network provider for %s %s/%s",
+				supervisorMachineCtx.Machine.GroupVersionKind(),
+				supervisorMachineCtx.Machine.Namespace,
+				supervisorMachineCtx.Machine.Name)
```

Comment:
If we hit this case, then it is not only missing for this object, but for all instead.

Reply:
With the change to update the NetworkProvider in VmopMachineService to a boolean, this particular check has become unnecessary.